File: c/gpt/p120.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    char *chunk1 = (char *)malloc(10);
    char *chunk2 = (char *)malloc(10);

    if (!chunk1 || !chunk2) {
        printf("Memory allocation failed\n");
        return 1;
    }

    strcpy(chunk1, "Hello");   // fits within 10 bytes
    strcpy(chunk2, chunk1);    // copy contents

    printf("%s\n", chunk2);

    free(chunk1);
    free(chunk2);
    return 0;
}

File: java/gpt/p120.java
public class MemoryCopy {
    public static void main(String[] args) {
        byte[] chunk1 = new byte[10];
        byte[] chunk2 = new byte[10];

        byte[] data = "Hello".getBytes();
        System.arraycopy(data, 0, chunk1, 0, data.length);
        System.arraycopy(chunk1, 0, chunk2, 0, chunk1.length);

        System.out.println(new String(chunk2).trim());
    }
}
